### DISCUSSION TITLE
/securing-azure-functions-existing-azure-ad-app/

### DISCUSSION THREAD

**Discussion by comment-migrator** [estruyf/eliostruyf-comments#567] [Comments]
Securing your Azure Functions with an existing Azure AD app
        
[https://www.eliostruyf.com/securing-azure-functions-existing-azure-ad-app/](https://www.eliostruyf.com/securing-azure-functions-existing-azure-ad-app/)

**Comment by comment-migrator** [comment]
```header
Originally posted by Anoop on 2021-04-16 02:14
```

Nice article Elio. Thank you for sharing.

I was reading about access tokens recently and [this Microsoft article](https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#payload-claims "https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens#payload-claims") mentions that in version 2.0 aad access token, the "aud" claim value will always be the GUID of the app.

To [get version 2.0 of access token](https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#accesstokenacceptedversion-attribute "https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-app-manifest#accesstokenacceptedversion-attribute") we need to set "accessTokenAcceptedVersion" to 2 in the manifest of the app registration.

With those settings does this mean, can the validateToken method in the azure function check for this GUID? (and also does this mean we can use this GUID in the allowed token audience? or have I got that wrong).

**Comment by comment-migrator** [comment]
```header
Originally posted by Anoop on 2021-05-05 17:54
```

To add to my previous comment, [this article](https://docs.microsoft.com/en-us/azure/app-service/configure-authentication-provider-aad#-enable-azure-active-directory-in-your-app-service-app "https://docs.microsoft.com/en-us/azure/app-service/configure-authentication-provider-aad#-enable-azure-active-directory-in-your-app-service-app") mentions that the configured client ID is always considered as an allowed audience. So does that mean, with version 2 of access tokens, we can skip validating audience in the Azure function?

**Comment by tatarianBarbarian** [comment]
Hello there! Should it work with Azure Directory **B2C**?
By some reason, I cannot make it working 😬

**Reply by estruyf** [reply]
Maybe the following documentation can help you out: https://learn.microsoft.com/en-us/azure/active-directory-b2c/configure-authentication-in-azure-web-app

**Reply by tatarianBarbarian** [reply]
Thank you for your response!

I checked, and it shows a message indicating that the App Registration created by Azure Functions supports only Microsoft Entra ID, not ADB2C. :c

Interestingly, using a manually created App Registration doesn’t work either.

So, it seems like custom middleware is the way to go. :)
